protocol/contracts/amo/AuraStaking.sol

```diff
@@ -89,14 +89,27 @@ contract AuraStaking is Ownable {
         emit RecoveredToken(token, to, amount);
     }
     
+    function isAuraShutdown() public view returns (bool) {
+        return booster.isShutdown() || booster.poolInfo(auraPoolInfo.pId).shutdown;
+    }
+
     function depositAndStake(uint256 amount) external onlyOperator {
-        bptToken.safeIncreaseAllowance(address(booster), amount);
-        booster.deposit(auraPoolInfo.pId, amount, true);
+        // Only deposit if the aura pool is open. Otherwise leave the BPT in this contract.
+        if (!isAuraShutdown()) {
+            bptToken.safeIncreaseAllowance(address(booster), amount);
+            booster.deposit(auraPoolInfo.pId, amount, true);
+        }
     }
 
     // withdraw deposit token and unwrap to bpt tokens
     function withdrawAndUnwrap(uint256 amount, bool claim, address to) external onlyOperatorOrOwner {
-        AMO__IBaseRewardPool(auraPoolInfo.rewards).withdrawAndUnwrap(amount, claim);
+        // Optimistically use BPT balance in this contract, and then try and unstake any remaining
+        uint256 bptBalance = bptToken.balanceOf(address(this));
+        uint256 toUnstake = (bptBalance >= amount) ? 0 : bptBalance - amount;
+        if (toUnstake > 0) {
+            AMO__IBaseRewardPool(auraPoolInfo.rewards).withdrawAndUnwrap(amount, claim);
```

Comment:
We could also only withdraw the difference above current bptBalance. wdyt about changing 108-110 to:
```
if (amount > bptBalance) 
  AMO__IBaseRewardPool(auraPoolInfo.rewards)
    .withdrawAndUnwrap(amount - bptBalance, claim);
```

Reply:
Doh - that was the intention - ln 110 should be:
`            AMO__IBaseRewardPool(auraPoolInfo.rewards).withdrawAndUnwrap(toUnstake, claim);`